### PR TITLE
Used make_unique instead of direct unique_ptr construction

### PR DESCRIPTION
**Pull request by alexeylysenko**
Use make_unique to create an object that isn’t shared (at least not yet), unless you need a custom deleter or are adopting a raw pointer from elsewhere.

### CODE REVIEW

File: src/libsync/configfile.cpp [hunk 1]
```diff
@@ -846,9 +846,9 @@ void ConfigFile::setClientVersionString(const QString &version)
     settings.setValue(clientVersionC(), version);
 }
 
-std::unique_ptr<QSettings> ConfigFile::settingsWithGroup(const QString &group, QObject *parent)
+std::unique_ptr<QSettings> ConfigFile::settingsWithGroup(const QString &group)
 {
-    std::unique_ptr<QSettings> settings(new QSettings(ConfigFile::configFile(), QSettings::IniFormat, parent));
```
Comment (erikjv) [inline src/libsync/configfile.cpp:851]: Why did you remove the `parent` argument?

Comment (TheOneRing) [reply src/libsync/configfile.cpp:851]: I just wanted to ask the same thing

Comment (erikjv) [reply src/libsync/configfile.cpp:851]: Whoops, that parent might result in a double delete when the parent gets destroyed before the unique ptr's constructor is run.

Comment (TheOneRing) [reply src/libsync/configfile.cpp:851]: hmm right.

I hope we properly use the function and don't keep a pointer in some member...

Comment (erikjv) [reply src/libsync/configfile.cpp:851]: I'll check that later this morning.

Comment (erikjv) [reply src/libsync/configfile.cpp:851]: We're fine, and it's good to get rid of the parent argument.

**Review by TheOneRing — APPROVED** [review]

File: src/libsync/configfile.cpp (hunk 1 above)
Comment (alexeylysenko) [reply src/libsync/configfile.cpp:851]: > I just wanted to ask the same thing

There was a mixed approach of object ownership: both parent and unique_ptr - this shouldn't be the case, they aren't interoperable. Double-delete problem will occur.
